File: 53-maximum-subarray/53-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        current = maxSub = nums[0]
        for i in range(1, len(nums)):
            current = max(nums[i], nums[i]+current)
            maxSub = max(maxSub, current)
        return maxSub
            

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int current = nums[0];
        int maxSub = nums[0];
        for (int i = 1; i < nums.length; i++) {
            current = Math.max(nums[i], nums[i]+current);
            maxSub = Math.max(maxSub, current);
        }
        return maxSub;
    }
}